controllers/components/kafka_topic.go

```diff
@@ -42,7 +46,52 @@ func (kt *KafkaTopic) Delete() (err error) {
 }
 
 func (kt *KafkaTopic) CheckDeviation() (problem, err error) {
-	return //TODO
+	name := kt.name
+	client := kt.KafkaTopics  // kt.KafkaTopics perform CRUD operations.
+
+	// leave this commented line for testing.
+	// topicIn, err := client.GetTopic("platform.inventory.events")
+	topicIn, err := client.GetTopic(name)
+
+	if err != nil {
+		return nil, errors.Wrap(err, 0)
+	}
+
+	if topicIn != nil {
+		var allTopics []unstructured.Unstructured
+		allTopics, err := client.GetAllTopics()
+		if err != nil {
+			return nil, errors.Wrap(err, 0)
+		}
+
+		// make the topic data accessible, which is private by design
+		topicInPtr, ok := topicIn.(*unstructured.Unstructured)
+		if !ok {
+			problem = fmt.Errorf("problem getting the topic %s details", topicIn)
+		}
+
+		// de-reference the topic pointer for comparison
+		topicInClear := *topicInPtr
+		fmt.Printf("Input topic name: %s\n", topicInClear.GetName())
+
+		if len(allTopics) > 0 {
+			for _, topic := range allTopics {
+
+				if topicInClear.GetName() == topic.GetName() {
+					if equality.Semantic.DeepEqual(topicInClear, topic) {
+						fmt.Printf("\nTopic named %s is identical.", topic.GetName())
+						return nil, nil
+					} else { 
+						fmt.Printf("\nTopic named %s NOT identical.", topic.GetName())
+						problem = fmt.Errorf("KafkaTopic named %s has changed.", topic.GetName())
```

Comment:
Here, you need to use a `return` statement:

`return fmt.Errorf("KafkaTopic named %s has changed.", topic.GetName()), nil`